code/modules/vehicles/mecha/combat/savannah_ivanov.dm

```diff
@@ -0,0 +1,277 @@
+			shake_for(SKYFALL_SINGLE_CHARGE_TIME-1, 10) // -1 gives space between the animates, so they don't interrupt eachother
+			spark_system.start()
+		if(SKYFALL_CHARGELEVEL_LAUNCH)
+			visible_message("<span class='danger'>[src] leaps into the air!</span>")
+			playsound(src, 'sound/weapons/gun/general/rocket_launch.ogg', 50, TRUE)
+	if(skyfall_charge_level != SKYFALL_CHARGELEVEL_LAUNCH)
+		INVOKE_ASYNC(src, .proc/skyfall_charge_loop, pilot)
+		return
+	for(var/mob/living/shaken in range(7, src))
+		shake_camera(shaken, 5, 5)
+	COOLDOWN_START(src, skyfall_cooldown, skyfall_cooldown_time)
+	var/turf/launch_turf = get_turf(src)
+	new /obj/effect/hotspot(launch_turf)
+	launch_turf.hotspot_expose(700, 50, 1)
+	new /obj/effect/skyfall_landingzone(launch_turf, src)
+	resistance_flags |= INDESTRUCTIBLE //not while jumping at least
+	mecha_flags |= QUIET_STEPS|QUIET_TURNS|CANNOT_INTERACT
+	phasing = "flying"
+	movedelay = 1
+	density = FALSE
+	layer = FLY_LAYER
+	animate(src, alpha = 0, time = 8, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+	animate(src, pixel_z = 400, time = 10, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL) //Animate our rising mech (just like pods hehe)
+	addtimer(CALLBACK(src, .proc/begin_landing, pilot), 2 SECONDS)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/shake_for(duration, amt)
+	var/offset = prob(50) ? -amt : amt
+	var/old_pixel_x = pixel_x
+	animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = -1, flags = ANIMATION_PARALLEL) //start shaking
+	addtimer(CALLBACK(src, .proc/stop_shaking, old_pixel_x), duration)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/stop_shaking(old_px)
+	if(skyfall_charge_level == SKYFALL_CHARGELEVEL_LAUNCH)
+		animate(src)
+	pixel_x = old_px
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/begin_landing(mob/living/pilot)
+	animate(src, pixel_z = 0, time = 10, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+	animate(src, alpha = 255, time = 8, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+	addtimer(CALLBACK(src, .proc/land, pilot), 10)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/land(mob/living/pilot)
+	visible_message("<span class='danger'>[src] crashes into the ground!</span>")
+	playsound(src, 'sound/effects/explosion1.ogg', 50, 1)
+	resistance_flags &= ~INDESTRUCTIBLE
+	mecha_flags &= ~(QUIET_STEPS|QUIET_TURNS|CANNOT_INTERACT)
+	phasing = initial(phasing)
+	movedelay = initial(movedelay)
+	density = TRUE
+	layer = initial(layer)
+	skyfall_charge_level = 0
+	update_icon_state()
+	for(var/turf/open/floor/tiled_turf in range(1, src))
+		tiled_turf.break_tile()
+	var/turf/landed_on = get_turf(src)
+	if(isclosedturf(landed_on))
+		landed_on.ScrapeAway()
+	for(var/mob/living/victim in landed_on)
+		to_chat(victim, "<span class='userdanger'>[src] lands on you from above!</span>")
+		if(victim.stat != CONSCIOUS)
+			victim.gib(FALSE, FALSE, FALSE)
+		else
+			victim.adjustBruteLoss(80)
```

Comment:
Make it damage not only living mobs, but all the objects and vehicles, else it's useless in mecha vs mecha combat. Maybe extend the damage from just one tile to 3x3 zone of landing

Reply:
Alright, I'll buff it